diff --git a/Classes/Routing/RoutingService.php b/Classes/Routing/RoutingService.php
index 25bf8f8be..e86b98453 100644
--- a/Classes/Routing/RoutingService.php
+++ b/Classes/Routing/RoutingService.php
@@ -484,6 +484,7 @@ class RoutingService implements LoggerAwareInterface
         }
 
         $queryParameterMap = $this->getQueryParameterMap();
+        $filterArray = array_values($filterArray);
         $newFilterArray = [];
         $defaultSeparator = $this->detectFacetAndValueSeparator((string)$filterArray[0]);
         // Collect parameter names and rename parameter if required
diff --git a/Tests/Unit/Routing/RoutingServiceTest.php b/Tests/Unit/Routing/RoutingServiceTest.php
index e46fb225a..dca1d1e70 100644
--- a/Tests/Unit/Routing/RoutingServiceTest.php
+++ b/Tests/Unit/Routing/RoutingServiceTest.php
@@ -160,6 +160,36 @@ class RoutingServiceTest extends SetUpUnitTestCase
         );
     }
 
+    #[Test]
+    public function testDeflateFilterQueryParameterWithSparseFilterKeysTest(): void
+    {
+        $routingService = $this->getRoutingService();
+
+        $queryParameters = [
+            'tx_solr' => [
+                'filter' => [
+                    1 => 'color:yellow',
+                    3 => 'taste:sour',
+                    5 => 'color:green',
+                ],
+            ],
+        ];
+        $expectedResult = [
+            'tx_solr' => [
+                'filter' => [
+                    'color:green,yellow',
+                    'taste:sour',
+                ],
+            ],
+        ];
+
+        self::assertTrue($routingService->shouldConcatQueryParameters());
+        self::assertEquals(
+            $expectedResult,
+            $routingService->concatQueryParameter($queryParameters),
+        );
+    }
+
     #[Test]
     public function testInflateFilterQueryParameterTest(): void
     {
