diff --git a/Classes/Routing/RoutingService.php b/Classes/Routing/RoutingService.php
index 25bf8f8be..e86b98453 100644
--- a/Classes/Routing/RoutingService.php
+++ b/Classes/Routing/RoutingService.php
@@ -484,6 +484,7 @@ class RoutingService implements LoggerAwareInterface
         }
 
         $queryParameterMap = $this->getQueryParameterMap();
+        $filterArray = array_values($filterArray);
         $newFilterArray = [];
         $defaultSeparator = $this->detectFacetAndValueSeparator((string)$filterArray[0]);
         // Collect parameter names and rename parameter if required
