--- a/Classes/Generator/SqlGenerator.php
+++ b/Classes/Generator/SqlGenerator.php
@@ -72,6 +72,7 @@
     {
         $indexes = [];
         $fields = [];
+        $foreignFields = [];
         $table = $tableDefinition->table;
         foreach ($tableDefinition->parentReferences as $parentReference) {
             $index = [];
@@ -89,6 +90,7 @@
             if (isset($parentTcaConfig['foreign_field'])) {
                 $foreignField = $parentTcaConfig['foreign_field'];
                 $index[] = $foreignField;
+                $foreignFields[] = $foreignField;
             }
             $indexes[] = $index;
         }
@@ -99,6 +101,12 @@
                 $sql[] = $sqlStatement;
             }
         }
+        foreach ($foreignFields as $foreignFieldName) {
+            $sqlStatement = 'CREATE TABLE `' . $table . '` (`' . $foreignFieldName . '` int(11) UNSIGNED DEFAULT \'0\' NOT NULL);';
+            if (!in_array($sqlStatement, $sql, true)) {
+                $sql[] = $sqlStatement;
+            }
+        }
         $indexes = array_map(fn(array $index): string => implode(', ', $index), $indexes);
         $indexes = array_unique($indexes);
         $indexes = array_values($indexes);
--- a/Tests/Unit/Generator/SqlGeneratorTest.php
+++ b/Tests/Unit/Generator/SqlGeneratorTest.php
@@ -76,6 +76,7 @@ public static function generateReturnsExpectedSqlStatementsDataProvider(): itera
             ],
             'expected' => [
                 "CREATE TABLE `foobar` (`fieldname` varchar(255) DEFAULT '' NOT NULL);",
+                "CREATE TABLE `foobar` (`foreign_table_parent_uid` int(11) UNSIGNED DEFAULT '0' NOT NULL);",
                 'CREATE TABLE `foobar` (KEY parent_uid (tablenames, fieldname, foreign_table_parent_uid));',
             ],
         ];
@@ -132,6 +133,8 @@ public static function generateReturnsExpectedSqlStatementsDataProvider(): itera
                 ],
             ],
             'expected' => [
+                "CREATE TABLE `foobar` (`foreign_field_1` int(11) UNSIGNED DEFAULT '0' NOT NULL);",
+                "CREATE TABLE `foobar` (`foreign_field_2` int(11) UNSIGNED DEFAULT '0' NOT NULL);",
                 'CREATE TABLE `foobar` (KEY parent_uid (foreign_field_1));',
                 'CREATE TABLE `foobar` (KEY parent_uid_2 (foreign_field_2));',
             ],
@@ -190,6 +193,8 @@ public static function generateReturnsExpectedSqlStatementsDataProvider(): itera
             ],
             'expected' => [
                 "CREATE TABLE `foobar` (`fieldname` varchar(255) DEFAULT '' NOT NULL);",
+                "CREATE TABLE `foobar` (`foreign_table_parent_uid` int(11) UNSIGNED DEFAULT '0' NOT NULL);",
+                "CREATE TABLE `foobar` (`alternative_foreign_field` int(11) UNSIGNED DEFAULT '0' NOT NULL);",
                 'CREATE TABLE `foobar` (KEY parent_uid (tablenames, fieldname, foreign_table_parent_uid));',
                 'CREATE TABLE `foobar` (KEY parent_uid_2 (alternative_foreign_table_field, fieldname, alternative_foreign_field));',
             ],
@@ -224,6 +229,7 @@ public static function generateReturnsExpectedSqlStatementsDataProvider(): itera
                 ],
             ],
             'expected' => [
+                "CREATE TABLE `foo_bar_collection` (`foreign_table_parent_uid` int(11) UNSIGNED DEFAULT '0' NOT NULL);",
                 'CREATE TABLE `foo_bar_collection` (KEY parent_uid (foreign_table_parent_uid));',
             ],
         ];
@@ -271,7 +277,9 @@ public static function generateReturnsExpectedSqlStatementsDataProvider(): itera
                 ],
             ],
             'expected' => [
+                "CREATE TABLE `collection2` (`foreign_table_parent_uid` int(11) UNSIGNED DEFAULT '0' NOT NULL);",
                 'CREATE TABLE `collection2` (KEY parent_uid (foreign_table_parent_uid));',
+                "CREATE TABLE `foo_bar_collection` (`foreign_table_parent_uid` int(11) UNSIGNED DEFAULT '0' NOT NULL);",
                 'CREATE TABLE `foo_bar_collection` (KEY parent_uid (foreign_table_parent_uid));',
             ],
         ];
