diff --git a/Classes/Middleware/VisualEditorDeclineButtonMiddleware.php b/Classes/Middleware/VisualEditorDeclineButtonMiddleware.php
--- a/Classes/Middleware/VisualEditorDeclineButtonMiddleware.php
+++ b/Classes/Middleware/VisualEditorDeclineButtonMiddleware.php
@@ -18,7 +18,7 @@
     public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
     {
         if ($this->shouldLoad($request)) {
-            $this->assetCollector->addJavaScriptModule('@webconsulting/webcon-easy-workspace/visual-editor-decline-button');
+            $this->assetCollector->addJavaScriptModule('@webconsulting/webcon-easy-workspace/visual-editor-decline-button.js');
         }
 
         return $handler->handle($request);
